sample/ingress-controller/README.md

```diff
@@ -0,0 +1,144 @@
+
+## Generate a TLS Cert
+
+```bash
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+    -out ingress-tls.crt \
+    -keyout ingress-tls.key \
+    -subj "/CN=demo.test.com/O=ingress-tls"
+```
+
+## Deploy Secrets-store CSI and the Azure Key Vault Provider
+https://github.com/Azure/secrets-store-csi-driver-provider-azure#install-the-secrets-store-csi-driver-and-the-azure-keyvault-provider
+
+## Deploy Ingress Controller
+
+**Create a namespace**
+
+```bash
+kubectl create ns ingress-test
+```
+
+**Helm install ingress-controller**
+
+```bash
+helm install stable/nginx-ingress --generate-name \
+    --namespace ingress-test \
+    --set controller.replicaCount=2 \
+    --set controller.nodeSelector."beta\.kubernetes\.io/os"=linux \
+    --set defaultBackend.nodeSelector."beta\.kubernetes\.io/os"=linux
+```
+
+## Deploy a SecretsProviderClass Resource
+
+- To provide identity to access key vault, refer to the following [section](https://github.com/Azure/secrets-store-csi-driver-provider-azure#provide-identity-to-access-key-vault).
+- Set the `tenantId` and `keyvaultName`
+
+```bash
+$ cat <<EOF | kubectl apply -f -
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: azure-tls
+spec:
+  provider: azure
+  secretObjects:
+  - secretName: ingress-tls-csi
+    type: kubernetes.io/tls
+    data: 
+    - objectName: ingresscert
+      key: tls.key
+    - objectName: ingresscert
+      key: tls.crt
+  parameters:
+    usePodIdentity: "false"
+    keyvaultName: "azkv"                        # the name of the KeyVault
+    objects: |
+      array:
+        - |
+          objectName: ingresscert
+          objectType: secret
+    tenantId: "xx-xxxxxxxx-xx"                  # the tenant ID of the KeyVault
+EOF
+```
+
+## Deploy Test Apps with Reference to Secrets Store CSI
+
+> NOTE: These apps reference a Secrets Store CSI volume and a `secretProviderClass` object created earlier. A Kubernetes secret `ingress-tls-csi` will be created by the CSI driver as a result of the app creation.
+
+```yaml
+      volumes:
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: "azure-tls"
+            nodePublishSecretRef:
+              name: secrets-store-creds
+```
+
+If not using [service principal mode](../../examples/service-principal-mode.md), remove the following snippet from [deployment-app-one.yaml](deployment-app-one.yaml) and [deployment-app-two.yaml](deployment-app-two.yaml)
+
+```yaml
+            nodePublishSecretRef:
+              name: secrets-store-creds
+```
+
+**Deploy the test apps**
+
+```bash
+kubectl apply -f sample/ingress-controller-tls/deployment-app-one.yaml -n ingress-test
+kubectl apply -f sample/ingress-controller-tls/deployment-app-two.yaml -n ingress-test
```

Comment:
s/ingress-controller-tls/ingress-controller

Reply:
right, updated the folder name to `ingress-controller-tls`